File: Array-Stack/Stack.py
class Stack:
    def __init__(self):
        self.arr = []
        self.size = 0
        
    def __init__(self, x):
        self.arr = []
        self.arr.append(x)
        self.size = 1;
        
    def push_back(self, x):
        self.arr.append(x)
        self.size += 1
        
    def pop_back(self):
        if self.size<=0:
            return -1
        else:
            temp = self.arr[-1]
            self.arr.pop()
            self.size -= 1
            return temp
        
    def Print(self):
        print(self.arr)
        

ob = Stack(5)
ob.Print()
ob.push_back(10)
ob.push_back(100)
ob.push_back(50)
ob.push_back(30)
ob.Print()
print(ob.pop_back())
print(ob.pop_back())
print(ob.pop_back())
print(ob.pop_back())
print(ob.pop_back())
print(ob.pop_back())
ob.Print()

File: Array-Stack/stack.java
class list{
  int [] arr;
  int s, len;
  list(){
    s = 0;
    len = 1;
    arr = new int[len];
  }

  void push(int x) {
    if(s==len) resize();
    arr[s] = x;
    s++;
  }

  int pop() {
    int temp = arr[s-1];
    s--;
    if(len/3>=s) resize();
    return temp;
  }

  void print() {
    System.out.print("[ ");
    for(int i=0; i<s; i++) System.out.print(arr[i]+ " ");
    System.out.println(']');
  }

  void resize() {
    len = s*2;
    int [] brr = new int[len];
    for(int i=0; i<s; i++) {
      brr[i] = arr[i];
    }
    arr = brr;
  }
}

public class stack{
  public static void main(String[] args) {
    list ob = new list();
    ob.push(5);
    ob.push(7);
    ob.push(9);

    ob.print();

    System.out.println(ob.pop());
    
    ob.print();
  }
}